### PR TITLE
Adding correct pre-allocation of frame encoded buffer

### PR DESCRIPTION
**Pull request by franz1981**

### CODE REVIEW

**Comment by franz1981** [comment]
This is making the behaviour a bit more Netty-idiomatic, saving expensive buffer (re)allocation on fly

File: src/main/java/io/strimzi/kproxy/codec/OpaqueFrame.java [hunk 1]
```diff
@@ -33,6 +33,12 @@ public abstract class OpaqueFrame implements Frame {
     public OpaqueFrame(ByteBuf buf, int length) {
         this.length = length;
         this.buf = buf.asReadOnly();
+        assert buf.readableBytes() == length;
+    }
+
+    @Override
+    public int estimateEncodedSize() {
+        return length + Integer.BYTES;
```
Comment (gunnarmorling) [inline src/main/java/io/strimzi/kproxy/codec/OpaqueFrame.java:41]: Why do we need to add the extra Integer bytes? Worth a comment in the code, I think.

**Review by gunnarmorling — APPROVED** [review]
@franz1981, this LGTM, one tiny question inline.

File: src/main/java/io/strimzi/kproxy/codec/OpaqueFrame.java (hunk 1 above)
Comment (franz1981) [reply src/main/java/io/strimzi/kproxy/codec/OpaqueFrame.java:41]: Out encoding requires length prefix (according to the encode method), but yes I can add a comment to make it clear -> or we can use length prefix channel handlers too (exists in netty but that's another extra hop)

Comment (gunnarmorling) [reply src/main/java/io/strimzi/kproxy/codec/OpaqueFrame.java:41]: Ah yes, I see. I'll take care of it and merge. Thanks!

**Pull request merged**